Lib/fontTools/ttLib/tables/_n_a_m_e.py

```diff
@@ -1197,6 +1188,19 @@ def visit(visitor, obj, attr, value):
     visitor.seen.add(value)
 
 
+@NameRecordVisitor.register(otTables.FeatureParamsCharacterVariants)
+def visit(visitor, obj):
+    for attr in ("FeatUILabelNameID", "FeatUITooltipTextNameID", "SampleTextNameID"):
+        value = getattr(obj, attr)
+        visitor.seen.add(value)
+    # also include the sequence of UI strings for individual variants, if any
+    if obj.FirstParamUILabelNameID == 0 or obj.NumNamedParameters == 0:
+        return
+    last_name_id = obj.FirstParamUILabelNameID + obj.NumNamedParameters - 1
+    if last_name_id >= 256 and last_namd_id <= 0x7FFF:
+        visitor.seen.update(range(obj.FirstParamUILabelNameID, last_name_id + 1))
```

Comment:
Is this check necessary, from the [spec](https://learn.microsoft.com/en-us/typography/opentype/spec/features_ae#tag-cv01--cv99) (emphasis mine):
> The values of featUiLabelNameId, featUiTooltipTextNameId and firstParamUiLabelNameId are **expected** to be in the font-specific name ID range (256–32767), though that is **not a requirement** in this Feature Parameters specification. The value of firstParamUiLabelNameId + numNamedParameters - 1 should not exceed 32767.

Reply:
> Is this check necessary 

which check exactly? `last_name_id >= 256` ?

Reply:
I agree with Khaled here. No need to check these things...

Reply:
The whole check. Do we check elsewhere that name IDs are less than 0x7FFF before using them?

Reply:
ok.. I simply copied the code verbatim from hb

Reply:
and there's even a typo! 🤣 

`last_namd_id <= 0x7FFF`

Reply:
We should drop the check from HarfBuzz then.

Reply:
I have removed the check. Ideally we'd have some test font to exercise this with. I know I could make one ad-hoc, but do you happen to know a font that contains these which I simply use?

Reply:
No, I don’t have or know such a font.

Reply:
@qxliu76 Can you remove this check from HB too? We don't enforce these kinds of things unless necessary.